File: ruby/collatz-conjecture/collatz_conjecture.rb
module CollatzConjecture

  def self.steps(number)
    raise ArgumentError if number <= 0
    count = 0
    until number == 1
      count += 1
      number = number.even? ? number / 2 : number * 3 + 1
    end
    count
  end

end

File: python/collatz-conjecture/collatz_conjecture.py
def steps(number, count = 0):
    if number <= 0:
        raise ValueError("Only positive integers are allowed")
    if number == 1:
        return count
    else:
        if number % 2 == 0:
            count += 1
            return steps(number / 2, count)
        else:
            count += 1
            return steps(number * 3 + 1, count)
